src/Components/GenericExpressCheckout/PaymentHandler/AbstractGenericExpressCheckoutPaymentHandler.php

```diff
@@ -42,10 +43,22 @@ protected function handleResponse(
         SalesChannelContext $salesChannelContext
     ): void {
         if (empty($response['status']) || $response['status'] === 'ERROR') {
-            throw new AsyncPaymentProcessException(
-                $transaction->getOrderTransaction()->getId(),
-                $this->translator->trans('PayonePayment.errorMessages.genericError')
-            );
+            if (class_exists(PaymentException::class)) {
+                throw PaymentException::asyncProcessInterrupted(
+                    $transaction->getOrderTransaction()->getId(),
+                    $this->translator->trans('PayonePayment.errorMessages.genericError')
+                );
+            } elseif (class_exists(AsyncPaymentProcessException::class)) {
+                // required for shopware version <= 6.5.3
+                // @phpstan-ignore-next-line
+                throw new AsyncPaymentProcessException(
+                    $transaction->getOrderTransaction()->getId(),
+                    $this->translator->trans('PayonePayment.errorMessages.genericError')
+                );
+            }
+
+            // should never occur. Just to be safe.
+            throw new \RuntimeException('payment process interrupted.');
```

Comment:
You can use the function `$this->createPaymentException` here.

Reply:
good argument. Thank you!